route/v1/types.go

```diff
@@ -136,6 +136,8 @@ type RouteSpec struct {
 	// Use the weight field in RouteTargetReference object to specify relative preference.
 	//
 	// +kubebuilder:validation:MaxItems=3
+	// +listType=map
+	// +listMapKey=name
```

Comment:
Is name really unique in this list, or is it the combination of name and kind?

Reply:
updated to 
```go
	// +listMapKey=name
	// +listMapKey=kind
```